zproject/backends.py

```diff
@@ -659,6 +661,18 @@ def sync_full_name_from_ldap(self, user_profile: UserProfile, ldap_user: _LDAPUs
                 raise ZulipLDAPException(e.msg)
             do_change_full_name(user_profile, full_name, None)
 
+    def sync_mail_from_ldap(
+        self, current_mail: str, ldap_user: _LDAPUser, realm: Realm
+    ) -> Optional[UserProfile]:
+        if (
+            hasattr(ldap_user, "dn")
+            and UserProfile.objects.filter(ext_auth_uid__ldap=ldap_user.dn, realm=realm).exists()
+        ):
+            user_profile = UserProfile.objects.get(ext_auth_uid__ldap=ldap_user.dn, realm=realm)
+            do_change_user_delivery_email(user_profile, current_mail)
```

Comment:
We certainly need to be logging what's happening here - mysterious changes of user emails resulting from this ldap syncing will be super hard to debug without that.